styleguide/component-select.scss

```diff
@@ -3,179 +3,190 @@
+$menu-size: 48px;
+$selector-fade-time: 150ms;
+$selector-fade-easing: linear;
 
 // component element needs to be position: relative for the selectors to display
 .component-selector-wrapper {
   position: relative;
 }
 
+// fade things behind the component
+.component-selector:before {
+  background: $selector-overlay;
+  content: '';
+  height: 100vh;
+  left: 0;
+  opacity: 0;
+  pointer-events: none;
+  position: fixed;
+  transition: opacity $selector-fade-time $selector-fade-easing;
+  top: 0;
+  width: 100vw;
+  z-index: -1;
+}
+
 // selector outlines
 .component-selector {
-  border-style: solid;
-  border-width: 1px;
-  // selector colors are set for the selected component by default,
-  // then overridden for the selected-parent. this is so we don't get a
-  // visual jump when animating them in
-  // (technically the parent DOES jump, but it's a lot less noticible than the
-  // selected component. plus, the colors are similar enough that you don't see it)
-  border-color: $selector-border;
-  // subtle shadow to create more visual distinction between selectors and the
-  // stuff behind them
-  box-shadow: 1px 1px 20px -11px $black;
-  height: calc(100% + #{$offset-height});
-  left: -$offset-width-half;
+  background: inherit;
+  height: 100%;
+  left: 0;
   opacity: 0;
   // because we're just setting opacity to show/hide, don't allow child selectors to be clicked
-  // note: selected-parent and selected both set this to `all` so current selectors can be clicked
+  // note: .selected sets this to `all` so current selectors can be clicked
   pointer-events: none;
   position: absolute;
-  top: -$offset-height-half;
-  transition: 50ms opacity linear; // fade out quickly
-  width: calc(100% + #{$offset-width});
-  z-index: -1; // should be less than any element in the component itself
+  top: 0;
+  transition: opacity $selector-fade-time $selector-fade-easing;
+  width: 100%;
 }
 
 // menus
 .component-selector-top,
 .component-selector-bottom {
+  @include component-toolbar-layer();
+
   background: $selector-bg;
-  border-color: transparent;
-  border-width: 1px;
-  border-style: solid;
-  right: -1px; // align borders
-  position: absolute;
+  border: 1px solid $selector-border;
+  height: $menu-size;
+  width: 100%;
+  right: 0;
+  position: fixed;
+
+  @media screen and (min-width: 600px) {
+    min-width: 100%;
+    position: absolute;
+    width: auto;
+  }
 }
 
-// place the top and bottom menus bisecting the outline vertically
 .component-selector-top {
-  border-color: $selector-border;
-  bottom: 100%;
+  top: 0;
+
+  @media screen and (min-width: 600px) {
+    bottom: calc(100% + #{$half-selector-offset});
+    top: auto;
+  }
 }
 
-// bottom menu is more offset, since it concerns actions that don't
-// affect the component itself (but rather things around it, e.g. adding new components)
 .component-selector-bottom {
-  border-color: $selector-border;
-  top: 100%;
+  top: calc(100% - 96px); // account for toolbar
```

Comment:
Is 96px a desktop+OS specific number? Anything we can tie this to that is a little more cross-platform?